monitoring/instrumented_mutex.cc

```diff
@@ -18,7 +18,9 @@ bool ShouldReportToStats(Env* env, Statistics* stats) {
 
 void InstrumentedMutex::Lock() {
   PERF_CONDITIONAL_TIMER_FOR_MUTEX_GUARD(db_mutex_lock_nanos,
-                                         stats_code_ == DB_MUTEX_WAIT_MICROS);
+                                         stats_code_ == DB_MUTEX_WAIT_MICROS &&
+                                             stats_ != nullptr &&
```

Comment:
what is the relationship between `db_mutex_lock_nanos` and non-null `stats_`?

Reply:
db_mutex_lock_nanos is part of PerfContext, not Statistics. I think Mark's report was more about the timer happening twice when both PerfContext and Statistics are enabled. In that case we could fix it by measuring time in db_mutex_lock_nanos, and using that value (subtracting its original value before the timer starts) to update Statistics::wait_time_micros